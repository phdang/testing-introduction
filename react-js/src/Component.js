import React, { useState } from 'react';

function Component() {
  const [count, setCount] = useState(0);

  const handleClick = () => {
    setCount(count + 1);
  };

  return (
    <div>
      <p>Count: {count}</p>
      <button className='btn' onClick={handleClick}>Increment</button>
    </div>
  );
}

export default Component;