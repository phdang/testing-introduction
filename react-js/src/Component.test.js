import React from 'react'
import Component from './Component'
import {render, fireEvent, screen} from '@testing-library/react'

test('matches snapshot', () => {
  const { asFragment } = render(<Component />)
  expect(asFragment()).toMatchSnapshot()
})

test('increments count on button click', () => {
  // Render the component
  render(<Component />)

  // Find the button element
  const button = screen.getByRole('button', { name: /Increment/i })
  
  // expect button has class name btn
  expect(button).toHaveClass('btn')

  // Assert initial count
  expect(screen.getByText(/Count: 0/i)).toBeInTheDocument() 

  // Simulate a click on the button
  fireEvent.click(button)

  // Assert count after click
  expect(screen.getByText(/Count: 1/i)).toBeInTheDocument() 
})
