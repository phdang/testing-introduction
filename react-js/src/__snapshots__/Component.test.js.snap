// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Count: 0
    </p>
    <button
      class="btn"
    >
      Increment
    </button>
  </div>
</DocumentFragment>
`;
